prov/tcp/src/tcpx_ep.c

```diff
@@ -209,18 +260,39 @@ static int tcpx_pep_sock_create(struct tcpx_pep *pep)
 		return -FI_EIO;
 	}
 
-	ret = tcpx_setup_socket(pep->sock);
-	if (ret) {
-		goto err;
-	}
+	if (ofi_addr_get_port(pep->info->src_addr) != 0) {
+		ret = tcpx_setup_socket(pep->sock);
+		if (ret) {
+			goto err;
+		}
+		ret = bind(pep->sock, pep->info->src_addr,
+		      (socklen_t) pep->info->src_addrlen);
+
+		if (ret) {
+			FI_WARN(&tcpx_prov, FI_LOG_EP_CTRL,
+				"failed to bind listener: %s\n",
+				strerror(ofi_sockerr()));
+			goto err;
+		}
+	} else {
```

Comment:
Please change this to
```
else if (port_range.high) {
    tcpx_bind_to_port()...
}
```
If a port range hasn't been specified, let the OS continue to pick it and avoid the overhead of the provider trying to find one.